mini_pupper_bringup/launch/bringup.launch.py

```diff
@@ -38,47 +57,71 @@ def generate_launch_description():
     bringup_launch_path = PathJoinSubstitution(
         [FindPackageShare('champ_bringup'), 'launch', 'bringup.launch.py']
     )
+    rviz_config_path = PathJoinSubstitution(
+        [description_package, 'rviz', 'urdf_viewer.rviz']
+    )
 
-    return LaunchDescription([
-        DeclareLaunchArgument(
+    robot_name = LaunchConfiguration("robot_name")
+    sim = LaunchConfiguration("sim")
+    hardware_connected = LaunchConfiguration("hardware_connected")
+    rviz = LaunchConfiguration("rviz")
+
+    declare_robot_name = DeclareLaunchArgument(
             name='robot_name',
             default_value='mini_pupper',
             description='Set robot name for multi robot'
-        ),
+        )
 
-        DeclareLaunchArgument(
+    declare_sim = DeclareLaunchArgument(
             name='sim',
             default_value='false',
             description='Enable use_sime_time to true'
-        ),
+        )
 
-        DeclareLaunchArgument(
+    declare_rviz = DeclareLaunchArgument(
             name='rviz',
-            default_value='false',
+            default_value='true',
             description='Run rviz'
-        ),
+        )
 
-        DeclareLaunchArgument(
+    declare_hardware_connected = DeclareLaunchArgument(
             name='hardware_connected',
             default_value='false',
             description='Set to true if connected to a physical robot'
-        ),
+        )
 
-        IncludeLaunchDescription(
+    rviz2_node = Node(
+            package="rviz2",
+            namespace="",
+            executable="rviz2",
+            name="rviz2",
+            arguments=["-d", rviz_config_path],
+            condition=IfCondition(rviz)
+        )
+
+    bringup_launch = IncludeLaunchDescription(
             PythonLaunchDescriptionSource(bringup_launch_path),
             launch_arguments={
-                "use_sim_time": LaunchConfiguration("sim"),
-                "robot_name": LaunchConfiguration("robot_name"),
-                "gazebo": LaunchConfiguration("sim"),
-                "rviz": LaunchConfiguration("rviz"),
-                "hardware_connected": LaunchConfiguration("hardware_connected"),
+                "use_sim_time": sim,
+                "robot_name": robot_name,
+                "gazebo": sim,
+                "rviz": "false",  # set always false to launch RViz2 with costom .rviz file
+                "hardware_connected": hardware_connected,
```

Comment:
It was renamed in CHAMP: https://github.com/chvmp/champ/blob/ros2/champ_bringup/launch/bringup.launch.py#L110. On the other hand, this flag is not used anymore. In a previous version, it was used as a conditional, which defined whether we needed to publish joint states.

Reply:
I added ToDos in my recent PR to CHAMP to address this parameter issue in the future.

Reply:
Thanks, I'll update the param name.

Reply:
Fixed in [f7d05af](https://github.com/mangdangroboticsclub/mini_pupper_ros/pull/40/commits/f7d05af892afcd6217358a6ef336687dee29e7d9)